kstatus/status/status.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package status
```

Comment:
add examples_test.go with testable examples for using the package, also a `doc.go` with the package docs.